### PR TITLE
🧪 #845 Updated Specimen Core Completion Logic

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

- Only Calculate Specimen Completion Stats based on DNA results

**Type of Change**

- [x] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [x] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [x] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/submission/submission-to-clinical/stat-calculator.ts [hunk 1]
```diff
@@ -85,14 +85,16 @@ const calcDonorCoreEntityStats = (
 
     if (tumorAndNormalExists) {
       coreStats[schemaNameToCoreCompletenessStat[clinicalType]] =
-        donor.specimens.map(sp => sp.clinicalInfo).filter(notEmpty).length / donor.specimens.length;
+        donor.specimens
+          .filter(specimen => specimen && specimen.sampleType?.includes('DNA'))
```
Comment (samrichca) [inline src/submission/submission-to-clinical/stat-calculator.ts:89]: I would change this to specify which sample types are included in the filter like so 

```
const dnaSampleTypes = ['ctDNA', ...];
...
specimen && specimen.sampleType && dnaSampleTypes.includes(specimen.sampleType);
```

Comment (demariadaniel) [reply src/submission/submission-to-clinical/stat-calculator.ts:89]: Sounds good, updated + tested. I'm guessing adding new DNA sample types is not something that happens often, i.e. this definition won't need to be dynamic.

Comment (samrichca) [reply src/submission/submission-to-clinical/stat-calculator.ts]: we'll have to update the code if sampleTypes get updated. it's good to not rely too much on naming conventions :)

File: src/submission/submission-to-clinical/stat-calculator.ts [hunk 2]
```diff
@@ -64,8 +64,10 @@ const coreClinialSchemaNamesSet = new Set<CoreClinicalSchemaName>(
   Object.keys(schemaNameToCoreCompletenessStat) as CoreClinicalSchemaName[],
 );
 
+const dnaSampleTypes = ['Amplified DNA', 'ctDNA', 'Other DNA enrichments', 'Total DNA'];
```
Comment (joneubank) [inline src/submission/submission-to-clinical/stat-calculator.ts:67]: Where did this list come from? Is it likely to be expanded and maintained further? It likely should move to a more general constants file.

File: src/submission/submission-to-clinical/stat-calculator.ts [hunk 3]
```diff
@@ -85,14 +87,19 @@ const calcDonorCoreEntityStats = (
 
     if (tumorAndNormalExists) {
       coreStats[schemaNameToCoreCompletenessStat[clinicalType]] =
-        donor.specimens.map(sp => sp.clinicalInfo).filter(notEmpty).length / donor.specimens.length;
+        donor.specimens
+          .filter(
+            specimen =>
+              specimen && specimen.sampleType && dnaSampleTypes.includes(specimen.sampleType),
+          )
+          .filter(notEmpty).length / donor.specimens.length;
```
Comment (joneubank) [inline src/submission/submission-to-clinical/stat-calculator.ts:95]: Can we add a comment here explaining why we are applying this filter?

Should we consider making a named filter function (like with `notEmpty`) that can be reused?
```ts
const dnaSpecimenFilter = (specimen?: Specimen): boolean => specimen && specimen.sampleType && dnaSampleTypes.includes(specimen.sampleType);
```

**Review by joneubank — APPROVED** [review]
Suggestions for helping future us understand these changes...

File: src/submission/submission-to-clinical/stat-calculator.ts (hunk 3 above)
Comment (demariadaniel) [reply src/submission/submission-to-clinical/stat-calculator.ts]: Comments are always important, added

Comment (demariadaniel) [reply src/submission/submission-to-clinical/stat-calculator.ts]: Reusable function added

File: src/submission/submission-to-clinical/stat-calculator.ts (hunk 2 above)
Comment (demariadaniel) [reply src/submission/submission-to-clinical/stat-calculator.ts]: I've moved dnaSampleTypes to the clinical-entities common definitions. Not really finding a better method for obtaining these values so we'll keep it as a string array for now.

**Pull request merged**